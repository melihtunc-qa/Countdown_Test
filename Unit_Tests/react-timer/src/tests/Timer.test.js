import React from 'react';
import { shallow } from 'enzyme';
import Timer from '../components/Timer/Timer';
import TimerButton from '../components/TimerButton/TimerButton';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Timer Component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Timer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('starts the timer', () => {
    const wrapper = shallow(<Timer />);
    wrapper.instance().startTimer();
    expect(wrapper.state('isOn')).toBe(true);
  });

  it('stops the timer', () => {
    const wrapper = shallow(<Timer />);
    wrapper.instance().stopTimer();
    expect(wrapper.state('isOn')).toBe(false);
  });

  it('resets the timer', () => {
    const wrapper = shallow(<Timer />);
    wrapper.instance().resetTimer();
    expect(wrapper.state('minutes')).toBe(25);
    expect(wrapper.state('seconds')).toBe(0);
  });
});
