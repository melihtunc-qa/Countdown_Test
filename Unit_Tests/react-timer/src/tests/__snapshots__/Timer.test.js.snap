// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Component renders without crashing 1`] = `ShallowWrapper {}`;
